docs/deployment/resource-providers/yarn.md

```diff
@@ -0,0 +1,200 @@
+
+# we assume to be in the root directory of the unzipped Flink distribution
+
+# (0) export HADOOP_CLASSPATH
+export HADOOP_CLASSPATH=`hadoop classpath`
+
+# (1) Start YARN Session
+./bin/yarn-session.sh --detached
+
+# (2) You can now access the Flink Web Interface through the URL printed in the last lines of the command output, or through the YARN ResourceManager web UI.
+
+# (3) Submit example job
+./bin/flink run ./examples/streaming/TopSpeedWindowing.jar
+
+# (4) Stop YARN session (replace the application id based on the output of the yarn-session.sh command)
+echo "stop" | ./bin/yarn-session.sh -yid application_XXXXX_XXX
+{% endhighlight %}
+
+Congratulations! You have successfully run a Flink application by deploying Flink on YARN.
+
+
+## Deployment Modes Supported by Flink on YARN
+
+For production use, we recommend deploying Flink Applications in the [Per-job or Application Mode]({% link deployment/index.md %}#deployment-modes), as these modes provide a better isolation for the Applications.
+
+### Application Mode
+
+Application Mode will launch a Flink cluster on YARN, where the main() method of the application jar gets executed on the JobManager in YARN.
+The cluster will shut down as soon as the application has finished. You can manually stop the cluster using `yarn application -kill <ApplicationId>` or by cancelling the Flink job.
+
+{% highlight bash %}
+./bin/flink run-application -t yarn-application ./examples/streaming/TopSpeedWindowing.jar
+{% endhighlight %}
+
+To unlock the full potential of the application mode, consider using it with the `yarn.provided.lib.dirs` configuration option
+and pre-upload your application jar to a location accessible by all nodes in your cluster. In this case, the 
+command could look like: 
+
+{% highlight bash %}
+./bin/flink run-application -t yarn-application \
+	-Dyarn.provided.lib.dirs="hdfs://myhdfs/my-remote-flink-dist-dir" \
+	hdfs://myhdfs/jars/my-application.jar
+{% endhighlight %}
+
+The above will allow the job submission to be extra lightweight as the needed Flink jars and the application jar
+are  going to be picked up by the specified remote locations rather than be shipped to the cluster by the 
+client.
+
+### Per-Job Cluster Mode
+
+The Per-job Cluster mode will launch a Flink cluster on YARN, then run the provided application jar locally and finally submit the JobGraph to the JobManager on YARN. If you pass the `--detached` argument, the client will stop once the submission is accepted.
+
+The YARN cluster will stop once the job has stopped.
+
+{% highlight bash %}
+./bin/flink run -m yarn-cluster --detached ./examples/streaming/TopSpeedWindowing.jar
+{% endhighlight %}
+
+With `bin/flink run -m yarn-cluster`, you can run the command line interface in YARN cluster mode, allowing to submit jobs to an ad-hoc YARN cluster. The command line options of the YARN session described below are then also available. They are prefixed with a `y` or `yarn` (for the long argument options).
```

Comment:
Maybe add an example how to interact with a per job cluster once it is deployed (e.g. when taking a savepoint or so).